plugin/src/main/java/org/opensearch/ml/cluster/MLSyncUpCron.java

```diff
@@ -154,6 +168,8 @@ public void run() {
                         log.error("Failed to sync model routing", ex);
                     })
                 );
+            // Undeploy expired models
+            undeployExpiredModels(expiredModels.keySet(), modelWorkerNodes);
```

Comment:
The current logic is very aggressive: if model expired on any node, we will undeploy it. It's possible that model expires on one node , but the other 99 nodes not yet. 

Suggestion: Let's gather model last access time and TTL from each node. Then create a method to check if model expired on all nodes, we will undeploy model. This will also be flexible for future change. For example if we want to undeploy model when expired on more than half nodes, we can just change the method.

Reply:
It costs too much to return lastAccessTime + TTL for each model and process all these info in the syncup Job, because we would need to construct a Map<modelId, Array of (lastAccessTime, TTL) for all nodes> and compute/compare for each entry. On the other hand, without adding new info, we can use the size of work nodes for expired model to do the same thing. If the model is expired in all deployed nodes, we will un-deploy.